src/geometry/euclidean/sphere.lean

```diff
@@ -15,6 +15,27 @@ in spheres in real inner product spaces and Euclidean affine spaces.
 
 * `mul_dist_eq_mul_dist_of_cospherical_of_angle_eq_pi`: Intersecting Chords Theorem (Freek No. 55).
 * `mul_dist_eq_mul_dist_of_cospherical_of_angle_eq_zero`: Intersecting Secants Theorem.
+* `mul_dist_add_mul_dist_eq_mul_dist_of_cospherical`: Ptolemy’s Theorem (Freek No. 95).
+
+TODO: The current statement of Ptolemy’s theorem works around the lack of a proper "cyclic polygon"
+concept in mathlib. (Or two such concepts, since both a strict version, where all vertices must be
+distinct, and a weak version, where consecutive vertices may be equal, would be useful in mathlib.
+Ptolemy should then use the weak one.) A natural statement would then use cyclic_polygon.
+
+An API needs to be built around that concept, which would include:
+- strict cyclic implies weak cyclic,
+- weak cyclic and consecutive points distinct implies strict cyclic,
+- weak/strict cyclic implies weak/strict cyclic for any subsequence,
+- any three points on a sphere are weakly or strictly cyclic according to whether they are distinct,
+- any number of points on a sphere intersected with a two-dimensional affine subspace are cyclic in
+  some order,
+- a list of points is cyclic if and only if its reversal is,
+- a list of points is cyclic if and only if any cyclic permutation is, while other permutations
+  are not when the points are distinct,
+- a point P as in the statement exists (and is unique) with weak/strict betweenness depending on
```

Comment:
For example, I'm unsure what is meant by "point P as in the statement."

Reply:
As in "Let ABCD be a cyclic quadrilateral whose diagonals cross at P. Then ..." which I gave as an informal description of the version of Ptolemy's theorem here. A point where the diagonals cross (currently expressed as certain angles equal to π, might be expressed in terms of a formal notion of "between" in future). Saying "where the diagonals cross" might be better for the TODO list than "as in the statement".